### PR TITLE
BUG: .loc assignment of datetime with tz is coercing to naive #11365

### LINKED ISSUE #11365: BUG: .loc assignment of datetime with tz is coercing to naive
xref #12502 (example there as well)
xref #14148 

```
In [3]: idx = pd.date_range('20130101',periods=3,tz='US/Eastern')

In [4]: df = DataFrame({'A' : [1,2,3]})

In [5]: df['B'] = idx

In [6]: df.dtypes
Out[6]: 
A                         int64
B    datetime64[ns, US/Eastern]
dtype: object

In [7]: df.loc[[True,False,True],'C'] = idx

In [8]: df
Out[8]: 
   A                         B                   C
0  1 2013-01-01 00:00:00-05:00 2013-01-01 05:00:00
1  2 2013-01-02 00:00:00-05:00                 NaT
2  3 2013-01-03 00:00:00-05:00 2013-01-02 05:00:00

# C should be the same time zone
In [9]: df.dtypes
Out[9]: 
A                         int64
B    datetime64[ns, US/Eastern]
C                datetime64[ns]
dtype: object
```

this is also realized to #11351 [4]

### PR DESCRIPTION
**Pull request by taeold**
closes #11365

### CODE REVIEW

**Comment by taeold** [comment]
sorry - i realize the fix is incomplete. will close and submit a new pull request soon.

File: pandas/core/common.py [hunk 1]
```diff
@@ -986,14 +986,15 @@ def _infer_fill_value(val):
...
         val = [val]
-    val = np.array(val,copy=False)
-    if is_datetimelike(val):
-        return np.array('NaT',dtype=val.dtype)
-    elif is_object_dtype(val.dtype):
-        dtype = lib.infer_dtype(_ensure_object(val))
+    v = np.array(val,copy=False)
+    if is_datetimelike(v):
+        if is_datetimetz(val):
+            return pd.DatetimeIndex(v, dtype=val.dtype)
```
Comment (jreback) [inline pandas/core/common.py]: shouldn't this be v? (and not val)

File: pandas/tests/test_indexing.py [hunk 2]
```diff
@@ -3446,6 +3446,13 @@ def test_loc_setitem_datetime(self):
             expected = DataFrame({'one' : [100.0,200.0]},index=[dt1,dt2])
             assert_frame_equal(df, expected)
 
+    def test_loc_setitem_datetimetz(self):
+        # GH 11365
+        idx = pd.date_range('20130101',periods=3,tz='US/Eastern')
+        df = DataFrame({'A': idx})
+        df.loc[[True,False,True],'B'] = idx
+        self.assert_equal(df['A'].dtype, df['B'].dtype)
```
Comment (jreback) [inline pandas/tests/test_indexing.py]: use assert_frame_equal and construct an expected frame

**Comment by jreback** [comment]
you don't need to close
just push more commits in this one

**Pull request closed without merging**